### ISSUE TITLE
run_maintenance() fails at drop_partition_time

### ISSUE THREAD

**Issue by bbcnova** [pgpartman/pg_partman#214]
Hello,

We are running Postgresql 10 (10.3-1.pgdg16.04+1) with pg_partman version 3.1.2 on Ubuntu 16.04. All partitions are using native partitioning and are based on time. 

I set the 'retention' value to 60 days in partman.part_config for all of our partitioned tables a bit over two months ago. However when I just went to check the partition status, I noticed that there are still partitions from 2/12 lingering around. 

Manually calling run_maintenance() fails with the following:

```
ERROR:  zero-length delimited identifier at or near """"
CONTEXT: PL/pgSQL function drop_partition_time(text,interval,boolean,boolean,text) line 235 at EXECUTE
PL/pgSQL function partman.run_maintenance(text,boolean,boolean,boolean) line 149 at assignment
DETAIL: 
HINT: 
CONTEXT: PL/pgSQL function drop_partition_time(text,interval,boolean,boolean,text) line 279 at RAISE
PL/pgSQL function partman.run_maintenance(text,boolean,boolean,boolean) line 149 at assignment
DETAIL: 
HINT: 
CONTEXT:  PL/pgSQL function partman.run_maintenance(text,boolean,boolean,boolean) line 358 at RAISE
```

If there is any other data that may be helpful just let me know. 

Thanks!

Edit: 

The following is the output from jobmon, which is also installed:

```
# select jobmon.check_job_status();
                                      check_job_status                                      
--------------------------------------------------------------------------------------------
 (3,FAILED_RUN,"PARTMAN DROP TIME PARTITION: PUBLIC.HISTORY","3 consecutive CRITICAL runs")
 (3,FAILED_RUN,"PARTMAN RUN MAINTENANCE","3 consecutive CRITICAL runs")
(2 rows)
```

**Comment by bbcnova** [comment]
Here is the output when running debug set to TRUE on the table referenced above in the jobmon output:

```
# select partman.run_maintenance('public.history', 'FALSE', 'TRUE', 'TRUE');
NOTICE:  run_maint: v_partition_expression: to_timestamp(clock)
NOTICE:  run_maint: parent_table: public.history, v_last_partition: history_p2018_04_28
ERROR:  zero-length delimited identifier at or near """"
CONTEXT: PL/pgSQL function drop_partition_time(text,interval,boolean,boolean,text) line 235 at EXECUTE
PL/pgSQL function partman.run_maintenance(text,boolean,boolean,boolean) line 149 at assignment
DETAIL: 
HINT: 
CONTEXT: PL/pgSQL function drop_partition_time(text,interval,boolean,boolean,text) line 279 at RAISE
PL/pgSQL function partman.run_maintenance(text,boolean,boolean,boolean) line 149 at assignment
DETAIL: 
HINT: 
CONTEXT:  PL/pgSQL function partman.run_maintenance(text,boolean,boolean,boolean) line 358 at RAISE
```

**Comment by bbcnova** [comment]
Figured it out. The issue was on my end and not pg_partman's. PEBKAC. 

Initially I had part_config set to move expired partitions to another schema that I had created. A while back I decided to just drop the partitions outright after the expire, so I ran something similar to:

```
update partman.part_config set retention_schema = '' where parent_table = '<parent_table_name>';
```

I assumed at the time that an empty value is treated the same as NULL in Postgresql. Apparently not (I am not a DBA).

I adjusted the config and changed '' to NULL via:

```
update partman.part_config set retention_schema = NULL where parent_table ='<parent_table_name';
```

Now calling run_maintenance() is successful and the old partitions were dropped as expected. 

Feel free to close this out.
